### DISCUSSION TITLE
Indicate unproto BBS that was received

### DISCUSSION THREAD

**Discussion by komacke** [mfncooper/paracon#8] [Ideas]
Typically the unproto path comes through in the received packet. The BBS received has a "*" next to it. I'm not sure where that gets inserted. I see it both from a kantronics TNC and from direwolf.

However that "*" does not come through in the paracon console.

Could we have that * come through? Or even better, color code the respective callsign. I'm starting to look at the code to see if I can propose a change but will take a little while to find where we lose the "*". :)

**Comment by dranch** [comment]
The "*" denotes the remote digipeater your station heard this UI packet from

**Reply by komacke** [reply]
Digipeater - that's right. I used the wrong term. In any case it would be nice to have flagged which digipeater was being received.

**Comment by komacke** [comment]
Hmm... I wonder if this is a limitation of AGWPE? I can see the regular expression used for parsing the message includes "*" in the callsign for the vias. I think maybe Direwolf's AGWPE does not pass that up - if I'm grabbing the message in the right place.

**Reply by mfncooper** [reply]
It's a limitation of Direwolf, rather than of AGWPE. Direwolf does not include the asterisks when it composes the text to send to AGWPE clients. I am not sure why that is. As you note, the asterisks are present on the Direwolf console, as expected.

The AGWPE spec is silent on this. There's no mention of a has-been-repeated indicator, and no examples of such an indicator. Thus we end up with different AGWPE servers behaving differently. Direwolf does not provide the asterisks, whereas ldsped does, for example.

Unfortunately, the data sent to Paracon in AGWPE frames does not include enough information for Paracon to be able to add back any missing asterisks.

I had thought there was an open issue against Direwolf for this, but I just had a quick look and did not see one.

**Comment by komacke** [comment]
Cool! Thanks for the follow up. I'll check on direwolf and open an issue.

It seems from reviewing the code that if direwolf adds the asterisk that it will "just" come through. It would be cool also to color code that one (making the received digipeater more clear). I'll leave it to your judgement to leave this request open or close it as won't do.

**Comment by komacke** [comment]
Here's a PR that would add color. Of course with direwolf it won't do anything different. Presumably it would with ldsped.

I tested with the last alpha character of a call in a known digipeater path. Switched the tested character to "*" before commit. So it's mostly tested, FWIW.

https://github.com/mfncooper/paracon/pull/9

**Comment by komacke** [comment]
It seems someone has provided a PR to direwolf to address this request. I built and tried it out. It passed up the * to paracon as expected. I'll try running my patched paracon and report the results in the PR.

Here's the direwolf PR:
https://github.com/wb2osz/direwolf/pull/530

**Comment by mfncooper** [comment]
The [Direwolf issue](https://github.com/wb2osz/direwolf/pull/530) was resolved some time ago, so closing this now.
